packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -1033,7 +1033,7 @@ export class Gtxn extends Op {
 
   execute (stack: TEALStack): void {
     this.assertUint8(BigInt(this.txIdx));
-    this.checkIndexBound(this.txIdx, this.interpreter.gtxs);
+    this.checkIndexBound(this.txIdx, this.interpreter.storageBranch.gtxs);
```

Comment:
`storage` should only contain the "blockhain" state, not current transactions. This should be in the runtime context or in interpreter context. So there are 2 options:
1. Interpreter has a reference to runtime, and uses runtime functions (`getTx`, `getAccount` etc...) - so it doesn't have a direct reference to a state and transaction context.
2. Similar to what you have now, we will create a `ctx` object in the interpreter (`interpreter.ctx`) which will be initialized by runtime when starting the interpreter (interpreter.execute). The ctx should be something like:

        ctx = {
          state,   // note: please rename storageBranch to state, interpreter shouldn't know if he operates on a branch or not. If we need are keeping a copy of the state in runtime, then in runtime we should use `stateBranch`.
          tx / gtx,
          currentAccount .... // not sure if this is needed.
        }
```

Reply:
ok